src/graphics/Explain.ts

```diff
@@ -69,320 +69,319 @@ export const EXPLAIN_DEATH = 'Surviving Death';
-            console.log('Tutorial: Complete task', task.text);
-            task.complete = true;
-            storage.set(getTutorialStorageKey(key), COMPLETE);
-            for (let nextTask of task.nextVisibleTasks) {
-                tutorialShowTask(nextTask);
-            }
-            updateTutorialChecklist();
-            isTutorialComplete();
-        } else {
-            console.error('No such tutorial task with key', key);
-        }
+  if (globalThis.doUpdateTutorialChecklist && (condition ? condition() : true)) {
+    const task = tutorialChecklist[key];
+    if (task) {
+      console.log('Tutorial: Complete task', task.text);
+      task.complete = true;
+      storage.set(getTutorialStorageKey(key), COMPLETE);
+      for (let nextTask of task.nextVisibleTasks) {
+        tutorialShowTask(nextTask);
+      }
+      updateTutorialChecklist();
+      isTutorialComplete();
+    } else {
+      console.error('No such tutorial task with key', key);
     }
+  }
 }
 globalThis.skipTutorial = async () => {
-    const yesSkip = await Jprompt({ text: 'skip tutorial detail', yesText: 'Yes', noBtnText: 'Cancel', noBtnKey: 'Escape', forceShow: true })
-    if (yesSkip) {
-        // Set all enemies as encountered:
-        globalThis.enemyEncountered = Object.keys(allUnits);
-        storage.set(storage.ENEMY_ENCOUNTERED_STORAGE_KEY, JSON.stringify(globalThis.enemyEncountered));
+  const yesSkip = await Jprompt({ text: 'skip tutorial detail', yesText: 'Yes', noBtnText: 'Cancel', noBtnKey: 'Escape', forceShow: true })
+  if (yesSkip) {
+    // Set all enemies as encountered:
+    globalThis.enemyEncountered = Object.keys(allUnits);
+    storage.set(storage.ENEMY_ENCOUNTERED_STORAGE_KEY, JSON.stringify(globalThis.enemyEncountered));
 
-        // Complete all tutorial tasks
-        for (let task of Object.keys(tutorialChecklist)) {
-            tutorialCompleteTask(task as keyof TutorialChecklist);
-        }
-        for (let key of Object.keys(explainMap)) {
-            storage.set(key, ALREADY_EXPLAINED);
-        }
+    // Complete all tutorial tasks
+    for (let task of Object.keys(tutorialChecklist)) {
+      tutorialCompleteTask(task as keyof TutorialChecklist);
     }
+    for (let key of Object.keys(explainMap)) {
+      storage.set(key, ALREADY_EXPLAINED);
+    }
+  }
 }
 export function tutorialShowTask(key: keyof TutorialChecklist) {
-    if (globalThis.doUpdateTutorialChecklist) {
-        const task = tutorialChecklist[key];
-        if (task) {
-            task.visible = true;
-            if (storage.get(getTutorialStorageKey(key)) === COMPLETE) {
-                tutorialCompleteTask(key);
-            }
-            setTutorialVisiblity(true);
-        } else {
-            console.error('No such tutorial task with key', key);
-        }
+  if (globalThis.doUpdateTutorialChecklist) {
+    const task = tutorialChecklist[key];
+    if (task) {
+      task.visible = true;
+      for (let popup of task.showExplainPopup) {
```

Comment:
I can't remember what's happening here?  Could you clarify with a comment please?

Reply:
Yes, I will add a comment 👍